### PR TITLE
Added pptx parser and image captioner

### PR DESCRIPTION
**Pull request by emptycrown**
Added a new file reader for .pptx files. Notably, it demonstrates the first use case of tokenizing non-text items (i.e. images) from within a file. Images inside the Powerpoint are captioned into text using `nlpconnect/vit-gpt2-image-captioning` and inserted into the Document.

### CODE REVIEW

**Review by jerryjliu — COMMENTED** [review]
nice!! is this going to get added to `DEFAULT_FILE_EXTRACTOR` in file/base.py as well?

**Comment by emptycrown** [comment]
> nice!! is this going to get added to `DEFAULT_FILE_EXTRACTOR` in file/base.py as well?

Yep good point, forgot about that at the end -- added

File: gpt_index/readers/file/slides_parser.py [hunk 1]
```diff
@@ -0,0 +1,119 @@
...
+        result = ""
+        for i, slide in enumerate(presentation.slides):
+            result += f"\n\nSlide #{i}: \n"
+            for shape in slide.shapes:
+                if hasattr(shape, "image"):
+                    image = shape.image
+                    # get image "file" contents
+                    image_bytes = image.blob
+                    # make up a name for the file
+                    image_filename = "tmp_image.%s" % image.ext
```
Comment (jerryjliu) [inline gpt_index/readers/file/slides_parser.py:110]: ubernit: let's keep formatting consistent throughout codebase e.g. f"tmp_image.{image.ext}"

**Review by jerryjliu — APPROVED** [review]
super excited for this

**Pull request merged**